C/eval.h

```diff
@@ -25,23 +30,25 @@
  *               input == NULL or UWORD input[ROUND_UWORD(inputSize)];
  *               if 'dag[len]' represents a Simplicity expression with primitives then 'NULL != env';
  */
-bool evalTCOExpression( bool *evalSuccess, UWORD* output, size_t outputSize, const UWORD* input, size_t inputSize
+bool evalTCOExpression( bool *evalSuccess, bool *evalCheck, UWORD* output, size_t outputSize, const UWORD* input, size_t inputSize
                       , const dag_node* dag, type* type_dag, size_t len, const txEnv* env
                       );
 
 /* Run the Bit Machine on the well-typed Simplicity program 'dag[len]'.
  *
- * If static analysis results determines the bound on memory allocation requirements exceed the allowed limits,
- * set '*evalSuccess' to 'false' and return 'true'.
  * If malloc fails, return 'false', otherwise return 'true'.
- * If during execution an 'assertr' or 'assertl' combinator fails, '*evalSuccess' is set to 'false'
- * Otherwise '*evalSuccess' is set to 'true'.
+ * If static analysis results determines the bound on memory allocation requirements exceed the allowed limits,
+ * set '*evalPass' to 'false'.
+ * If during execution an 'assertr' or 'assertl' combinator fails, '*evalPass' is set to 'false'.
+ * If after execution the anti-DOS checks fail, '*evalPass' is set to 'false'.
+ * Otherwise '*evalPass' is set to 'true'.
  *
- * Precondition: NULL != evalSuccess
+ * Precondition: NULL != evalPass
```

Comment:
In 2d58ec81f4a848b7f7083e3e85ba85e78545b7b6:

I think this comment is out of date -- you can pass `NULL` for `evalPass` and it'll do the right thing (well, it will ignore success or failure entirely :)).

Reply:
Obsolete.